Makefile

```diff
@@ -4,7 +4,12 @@
 PROJECT_NAME ?= upjet-provider-template
 PROJECT_REPO ?= github.com/upbound/$(PROJECT_NAME)
 
-export TERRAFORM_VERSION ?= 1.2.1
+export TERRAFORM_VERSION ?= 1.5.7
```

Comment:
This bumps the default terraform version to the latest v1.5.x version, which is still under MPL license: https://github.com/hashicorp/terraform/blob/v1.5.7/LICENSE

This is not a requirement in this PR, but I thought it would be nice to update the default version to something new, since we are seeing community members doing that themselves already 😉